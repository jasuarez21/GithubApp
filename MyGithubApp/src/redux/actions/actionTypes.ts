export default {
    GET_USER: 'GET_USER',
    ERROR: 'ERROR'
}