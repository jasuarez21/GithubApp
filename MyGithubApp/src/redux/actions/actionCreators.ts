import axios from 'axios';
import actionTypes from './actionTypes';
import {Dispatch} from 'redux';

export function getUsers(user: string){
    console.log(user)
    return async (dispatch: Dispatch) => {
        try{
            const {data} = await axios.get('the url')
            dispatch({
                type: actionTypes.GET_USER,
                data,
            })
        } catch (error) {
            dispatch({
                type: actionTypes.ERROR
            })
        }
    }
}