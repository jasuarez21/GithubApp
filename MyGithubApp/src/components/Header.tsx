import React, {useState} from 'react';
import {useSelector, useDispatch} from 'react-redux';
import {StackScreenProps} from '@react-navigation/stack';
import {TouchableHighlight, View, Text, TextInput, SafeAreaView, ScrollView,  StyleSheet, Image} from 'react-native';
import {getUsers} from '../redux/actions/actionCreators';


interface Props extends StackScreenProps<any, any> {}

const Header = ({navigation}: Props) => {
    const user = useSelector((store: any) => store.user);
    const dispatch = useDispatch();
    let [userSearch, setUserSearch] = useState('');
    function userSearched(user){
        setUserSearch(user);
        dispatch(getUsers(user))
    }
    return (
        <View style={styles.headerContainer}>
            <TextInput
            style={styles.input}
            testID='searchInput'
            placeholder="Introduce el usuario"
            autoCapitalize="none"
            onChangeText={(user: string) => {userSearched(user)}}
            defaultValue={userSearch}
            value={userSearch}
            />
        </View>
    )
}

const styles = StyleSheet.create({
    headerContainer: {
        display: 'flex',
        flexDirection: 'row',
        alignSelf: 'center',
        marginTop: 15
    },
    input: {
        backgroundColor: 'white',
        width: 250,
        height: 50,
        marginTop: 10,
        marginLeft: 20,
        borderRadius: 25
    }
})

export default Header;