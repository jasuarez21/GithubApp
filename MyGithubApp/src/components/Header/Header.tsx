import React, {useState} from 'react';
import {useSelector, useDispatch} from 'react-redux';
import {StackScreenProps} from '@react-navigation/stack';
import {TouchableHighlight, View, Text, TextInput, SafeAreaView, ScrollView,  StyleSheet, Image} from 'react-native';
import {searchUser} from '../../redux/actions/actionCreators';


interface Props extends StackScreenProps<any, any> {}

const Header = ({navigation}: Props) => {
    const user = useSelector((store: any) => store.user);
    const dispatch = useDispatch();
    let [userSearch, setUserSearch] = useState('');
    function userSearched(user){
        setUserSearch(user);
        dispatch(searchUser(user))
    }
    return (
        <View style={styles.headerContainer}>
            <Image
                style={styles.logo}
                source={{
                    uri: "https://i.postimg.cc/W4pH9FN9/icons8-github-50.png",
                }}
            />  
            <TextInput
            style={styles.input}
            testID='searchInput'
            placeholder="Introduce el usuario"
            autoCapitalize="none"
            onChangeText={(user: string) => {userSearched(user)}}
            defaultValue={userSearch}
            value={userSearch}
            />
        </View>
    )
}

const styles = StyleSheet.create({
    headerContainer: {
        display: 'flex',
        flexDirection: 'row',
        alignSelf: 'center',
        marginTop: 15
    },
    logo: {
        height: 50,
        width: 50,
        marginTop: 10,
        marginLeft: -40
    },
    input: {
        backgroundColor: '#caf0f8',
        width: 250,
        height: 50,
        marginTop: 10,
        marginLeft: 20,
        borderRadius: 25,
        borderWidth: 1,
        borderColor: '#023e8a',
        padding: 10
    }
})

export default Header;