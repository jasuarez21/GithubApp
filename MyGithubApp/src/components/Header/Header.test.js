import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../redux/actions/actionCreators';
import { Provider } from 'react-redux';
import Header from './Header';
import actionTypes from '../../redux/actions/actionTypes';

jest.mock('../../redux/actions/actionCreators');

describe('When invoked a Header component', () => {
    const mockStore = ConfigureStore();
    const store = mockStore({
        user : [{
            login: 'Pepe'
        },
        {
            login: 'Juanito' 
        }]
    })
    let route;
    let navigation;
    beforeEach(() => {
        jest.spyOn(actions, 'searchUser').mockReturnValueOnce({ type: actionTypes.GET_USERS });
        dispatch = jest.fn();
    })
    test('Should render the header page', () => {
        const header = render(<Provider store={store}><Header /></Provider>)
        expect(header).toMatchSnapshot()
    })
    test('Should search and user when i write in input', async () => {
        const { getByTestId } = render(<Provider store={store}><Header /></Provider>);
        const searchInput = getByTestId('searchInput');
        const userSearched = jest.fn();
        await fireEvent.changeText(searchInput, userSearched);
        expect(actions.searchUser).toHaveBeenCalled();
    })
})
