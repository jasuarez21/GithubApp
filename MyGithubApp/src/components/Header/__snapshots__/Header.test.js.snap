// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a Header component Should render the header page 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "display": "flex",
      "flexDirection": "row",
      "marginTop": 15,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://i.postimg.cc/W4pH9FN9/icons8-github-50.png",
      }
    }
    style={
      Object {
        "height": 50,
        "marginLeft": -40,
        "marginTop": 10,
        "width": 50,
      }
    }
  />
  <TextInput
    autoCapitalize="none"
    defaultValue=""
    onChangeText={[Function]}
    placeholder="Introduce el usuario"
    style={
      Object {
        "backgroundColor": "#caf0f8",
        "borderColor": "#023e8a",
        "borderRadius": 25,
        "borderWidth": 1,
        "height": 50,
        "marginLeft": 20,
        "marginTop": 10,
        "padding": 10,
        "width": 250,
      }
    }
    testID="searchInput"
    value=""
  />
</View>
`;
